components/blitz/src/omero/gateway/LoginCredentials.java

```diff
@@ -51,15 +57,74 @@
     /** ID of the group if not the default group of the user should be used */
     private long groupID = -1;
 
+    /** The connection argument.*/
+    private ImmutableList<String> args;
+
     /**
      * Creates a new instance
      */
     public LoginCredentials() {
         user = new UserCredentials();
         server = new ServerInformation();
+        args = null;
     }
 
     /**
+     * Creates a new instance.
+     * @param args The connection arguments.
+     */
+    public LoginCredentials(String[] args)
+    {
+        this();
+        if (args == null) {
+            throw new IllegalArgumentException("No connection arguments");
+        }
+        this.args = ImmutableList.copyOf(args);
+    }
+
+    /**
+     * Creates a new instance with the given credentials and default port
+     * 
+     * @param username
+     *            The username
+     * @param password
+     *            The password
+     * @param host
+     *            The server hostname
+     */
+    public LoginCredentials(String username, String password, String host) {
+        this(username, password, host, omero.constants.GLACIER2PORT.value);
```

Comment:
I was wondering about the case where the OMERO ports have been completely changed from their defaults by configuration. (For instance, maybe we can see if `omero.port` was set in ice.config or somesuch?)